### PR TITLE
Add support for pull request URLs

### PR DESCRIPTION
**Pull request by agrare**
Allow any of the test/core/gem repos to be specified by its URL.

### CODE REVIEW

File: lib/manageiq/cross_repo/repository.rb [hunk 1]
```diff
@@ -45,6 +45,21 @@ def parse_identifier(identifier, server)
         repo = path.basename.to_s
         ref  = Dir.chdir(path) { `git rev-parse HEAD`.chomp }
         sha  = ref
+      elsif identifier.start_with?(server)
+        identifier = URI.parse(identifier).path
+        org_and_repo, pr = identifier.split("/pull/")
+        _, org, repo = org_and_repo.split("/")
```
Comment (agrare) [inline lib/manageiq/cross_repo/repository.rb:51]: The path has a leading `/` e.g. `/ManageIQ/manageiq/pull/1234` so I could either strip it out then split, or just split and ignore the first component

**Comment by Fryguy** [comment]
Can you also support branch URLs and commit URLs?

branch: `https://github.com/JoeSmith/manageiq/tree/ivanchuk`
commit: `https://github.com/JoeSmith/manageiq/commit/abcdef123456789`

File: lib/manageiq/cross_repo/repository.rb (hunk 1 above)
Comment (Fryguy) [reply lib/manageiq/cross_repo/repository.rb:51]: This is fine.

**Comment by agrare** [comment]
> Can you also support branch URLs and commit URLs?

Yeah sure thing

**Comment by Fryguy** [comment]
As discussed let's do branches and commits in a follow up

**Pull request merged**